### PR TITLE
client: add dial_server_timeout

### LINKED ISSUE #2761: set default connect timeout in frpc

### PR DESCRIPTION
**Pull request by fatedier**
Fix #2761

### CODE REVIEW

File: test/e2e/basic/client_server.go [hunk 1]
```diff
@@ -268,5 +268,4 @@ var _ = Describe("[Feature: Client-Server]", func() {
 			})
 		}
 	})
-
```
Comment (bingtianbaihua) [inline test/e2e/basic/client_server.go:271]: What's this ?

Comment (fatedier) [reply test/e2e/basic/client_server.go:271]: Empty line

**Comment by bingtianbaihua** [comment]
Is it necessary to update in README.md doc ?

**Comment by fatedier** [comment]
> Is it necessary to update in README.md doc ?

No, it's not a notable feature. Users can find all configs here https://github.com/fatedier/frp#configuration-files .

**Pull request merged**